temporal-sdk/src/main/java/io/temporal/internal/statemachines/WorkflowStateMachines.java

```diff
@@ -705,6 +735,20 @@ private void handleNonStatefulEvent(HistoryEvent event, boolean hasNextEvent) {
       case EVENT_TYPE_WORKFLOW_EXECUTION_CANCEL_REQUESTED:
         callbacks.cancel(event);
         break;
+      case EVENT_TYPE_WORKFLOW_EXECUTION_UPDATE_ADMITTED:
+        WorkflowExecutionUpdateAdmittedEventAttributes admittedEvent =
+            event.getWorkflowExecutionUpdateAdmittedEventAttributes();
+        Message msg =
+            Message.newBuilder()
+                .setId(admittedEvent.getRequest().getMeta().getUpdateId() + "/request")
+                .setProtocolInstanceId(admittedEvent.getRequest().getMeta().getUpdateId())
+                .setEventId(event.getEventId())
+                .setBody(Any.pack(admittedEvent.getRequest()))
+                .build();
+        if (replaying && acceptedUpdates.remove(msg.getProtocolInstanceId()) || !replaying) {
+          messages.add(msg);
```

Comment:
(Note to check I'm understanding; correct me if not!) OK so here, you've already peeked forwards, and hence on encountering `Admitted` know whether it's going to be `Accepted` in the future. In contrast, in the `sdk-core` implementation, we do not peek forwards for this, and instead hold the durably `Admitted` info for use when we encounter an `Accepted`.

Reply:
Hm thinking about it more i think I can make Javas implementation align more with Core and Go's. going to resolve all the comments related how admitted is handled since it will all change slightly.